static/js/subs.js

```diff
@@ -725,20 +725,51 @@ function ajaxSubscribe(stream) {
 
 function ajaxUnsubscribe(sub) {
     // TODO: use stream_id when backend supports it
-    return channel.del({
-        url: "/json/users/me/subscriptions",
-        data: {subscriptions: JSON.stringify([sub.name]) },
-        success: function () {
-            $(".stream_change_property_info").hide();
-            // The rest of the work is done via the unsubscribe event we will get
-        },
-        error: function (xhr) {
-            ui_report.error(i18n.t("Error removing subscription"), xhr,
-                            $(".stream_change_property_info"));
-        },
-    });
+    if (sub.invite_only) {
+        var unsubscribe_stream_modal = templates.render('unsubscribe_stream', {stream_name: sub.name});
+        $(".subscription_settings").append(unsubscribe_stream_modal);
```

Comment:
It's not showing when opened from stream action popover menu because the parent of `.subscription_settings` is hidden so modal doesn't appear.
One possible fix here is to add this rendered HTML to a container which would be another sibling div of `subscriptions_table` whose HTML you will change everytime modal is opened by `.html()`. This is just a possible solution feel free to change this according to your logic or whatever you find appropriate as there will be many edge cases.

Reply:
I'm worried to not able to understand what you're trying to say. 

Would you please elaborate @shubhamdhama

Reply:
So let me explain you in an easy way:
```
<div class="subscriptions">
    <div id="subscriptions_table"><div id="subscription_overlay" class="overlay  new-style" data-overlay="subscriptions" aria-hidden="true">
    <div class="flex overlay-content">
        <div class="subscriptions-container modal
```
This is the structure of overlay, so when stream settings overlay is closed anything inside it isn't visible and since your modal is appended to `subscription_settings` which is further at a lower level of this modal, your modal doesn't appear. Means your modal won't appear unless stream settings overlay is opened. To deal with this you've to bring your modal outside of this overlay structure so that you can make it show in any situation (Though I think setting CSS property `!important` manually could work but then you've to have to handle various cases so better would be to just create a container outside the overlay and deal with this there). Now reading above comment will make better sense. Try grepping keywords I've mentioned then you may get a better idea what is happening here.

Reply:
If I use HTML instead of handlebars, the problem doesn't ever come in the first place.
Can use HTML or is there any specific use of using handlebars?

Reply:
Sorry I don't get it what you mean by "HTML instead of handlbars", `.handlebars` are also HTML (templates). Can you please discuss your problem/alternative at #issues or #developement?

If you think your PR isn't reviewed for a long time you should ask for code review at #code review or ask help at #issues if got stuck anywhere.

Reply:
In the above comment, you mentioned about ```container`` 
"... so better would be to just create a container outside the overlay and deal with this there."
Could you please specify about which container were you talking about?